tests/kernel/timer/timer_api/src/timer_convert.c

```diff
@@ -287,8 +281,8 @@ static void test_conversion(struct test_rec *t, uint64_t val)
 	zassert_true(diff <= maxdiff && diff >= mindiff,
 		     "Convert %llu (%llx) from %u Hz to %u Hz %u-bit %s\n"
 		     "result %llu (%llx) diff %lld (%llx) should be in [%lld:%lld]",
-		     val, val, from_hz, to_hz, t->precision, round_s[t->round],
-		     result, result, diff, diff, mindiff, maxdiff);
+		     val, val, from_hz, to_hz, t->precision, t->round, result, result, diff, diff,
```

Comment:
It is not clear to me why round_s would be showing up as unused. However, if it is removed, then the format string above would have to be updated as t->round is an integer, not a string.

Reply:
I've just opened #89482 to add validation of strings used in zasserts. Maybe that will fix unused variables case as well as it seems that for some reason formatted string used for zassert is not evaluated when unused variable warning is raised.